plugins/civmodcore-paper/src/main/java/vg/civcraft/mc/civmodcore/players/PlayerNames.java

```diff
@@ -0,0 +1,48 @@
+package vg.civcraft.mc.civmodcore.players;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.StringUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerLoginEvent;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+public final class PlayerNames {
+	private static final Set<String> names = Collections.synchronizedSet(new HashSet<>());
```

Comment:
```suggestion
	private static final Set<String> names = new HashSet<>();
```
No async going on here anymore